src/Tes.Runner/Docker/DockerExecutor.cs

```diff
@@ -186,35 +187,16 @@ private async Task PullImageWithRetriesAsync(string imageName, string? tag, Auth
                     new Progress<JSONMessage>(message => logger.LogDebug(message.Status))));
         }
 
-        private async Task DeleteAllImagesAsync()
+        private async Task DeleteImageAsync(string imageName)
         {
             try
             {
-                var images = await dockerClient.Images.ListImagesAsync(new ImagesListParameters { All = true });
-
-                if (images is null)
-                {
-                    return;
-                }
-
-                foreach (var image in images)
-                {
-                    try
-                    {
-                        await dockerClient.Images.DeleteImageAsync(image.ID, new ImageDeleteParameters { Force = true });
-                        logger.LogInformation(@"Deleted Docker image with ID: {ImageID}", image.ID);
-                    }
-                    catch (Exception e)
-                    {
-                        logger.LogInformation(@"Failed to delete image with ID: {ImageID}. Error: {ErrorMessage}", image.ID, e.Message);
-                        throw;
-                    }
-                }
+                await dockerClient.Images.DeleteImageAsync(imageName, new ImageDeleteParameters { Force = true });
+                logger.LogInformation(@"Deleted Docker image {ImageName}", imageName);
             }
             catch (Exception e)
             {
-                logger.LogError(e, "Exception in DeleteAllImagesAsync");
-                throw;
+                logger.LogWarning(@"Failed to delete image {ImageName}. Error: {ErrorMessage}", imageName, e.Message);
```

Comment:
Have you checked the logs to ensure this is not written?  How did you test to see if the image was actually deleted?

Reply:
Yeah I downloaded the tes runner logs and confirmed it printed " Deleted Docker image ". Since we're only deleting the task image, actually I should probably add the throw back into this catch.